docs/guides/routing.mdx

```diff
@@ -0,0 +1,45 @@
+---
+description: Learn how Clerk handles routing in your application.
+---
+
+# Routing in Clerk
```

Comment:
I'm not sure this belongs under `guides`. There's no concrete outcome or goal here, it's more-so an educational piece on some of Clerk's internals and how they relate to the user-facing props. 🤔

At a quick glance though I'm not sure where else it would go. Maybe we'd want a new section on "How Clerk works" or something.

Reply:
it will probably have a better home after the IA

cc @kylemac